neqo-common/src/timer.rs

```diff
@@ -202,15 +202,9 @@ impl<T> Timer<T> {
             }
         }
 
-        let idx = self.bucket(0);
-        for i in idx..self.items.len() {
-            let res = maybe_take(&mut self.items[i], until);
-            if res.is_some() {
-                return res;
-            }
-        }
-        for i in 0..idx {
-            let res = maybe_take(&mut self.items[i], until);
+        for i in self.cursor..(self.cursor + self.delta(until)) {
+            let i = i % self.items.len();
+            let res = maybe_take(&mut self.items[i], until)?;
```

Comment:
If you do this, then the function can be inlined.

However, this will regress one aspect of performance to save another.  The reason for two loops is to avoid the modulus operation inside the loop.  I think that what you want is more complex yet.

You want the first loop to be `self.bucket(0)..min(self.items.len(), self.bucket(0) + self.delta(until))` and the second to be `0..min(self.bucket(0), self.bucket(0) + self.delta(until) - self.items.len())`  ... I think.

Reply:
Oh, and I should have said: this is an obvious win, so it's worth doing.  I just want to make sure that we get the whole win.